openedx/openedx-platform
# feat: add a POST endpoint for listing courses

## Description

This pull request introduces functionality to filter courses using a POST request in the course API. The reason for this is to be able to able to use a filter larger than allowed in GET query params.

The key changes include adding a new test for POST request filtering, modifying the `get_queryset` method to handle POST data, and adding a POST method to the view.

## Testing instructions

1. Make a POST request to `/api/courses/v1/courses/` to query for courses.
2. Ensure the results are the same for the GET request.

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.
